pkg/storage/etcd/etcd_helper.go

```diff
@@ -711,3 +710,21 @@ func (h *etcdHelper) addToCache(index uint64, obj runtime.Object) {
 		metrics.ObserveNewEntry()
 	}
 }
+
+func convertToStorageErrorIfPossible(err error, key string, rv uint64) error {
```

Comment:
is this an etcd specific func? or we might add other potential kv error types into switch cases?